collector/infiniband_linux.go

```diff
@@ -0,0 +1,167 @@
+// Copyright 2017 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build linux
+// +build !noinfiniband
+
+package collector
+
+import (
+	"errors"
+	"path/filepath"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/common/log"
+)
+
+const infinibandPath = "class/infiniband"
+
+var (
+	infinibandNoDevicesFound = errors.New("InfiniBand / No devices detected")
+	infinibandNoPortsFound   = errors.New("InfiniBand / No ports detected")
+)
+
+type infinibandCollector struct {
+	metricDescs map[string]*prometheus.Desc
+	counters    map[string]string
+}
+
+func init() {
+	Factories["infiniband"] = NewInfiniBandCollector
+}
+
+func NewInfiniBandCollector() (Collector, error) {
+	var i infinibandCollector
+
+	// Filenames of all InfiniBand counter metrics including a detailed description.
+	i.counters = map[string]string{
+		"link_downed":            "Number of times the link failed to recover from an error state and went down",
+		"link_error_recovery":    "Number of times the link successfully recovered from an error state",
+		"multicast_rcv_packets":  "Number of multicast packets received (including errors)",
```

Comment:
I totally agree. At this point I am just curious if it's okay to go back and "fix" metrics names since that would break metrics for previous timeseries when the node_exporter is upgraded.

I was working on a tool at one point that would lint metric names and tell you what the conventions were, but I haven't gotten back to it recently.